product-mini/platforms/linux/CMakeLists.txt

```diff
@@ -63,7 +63,7 @@ set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wformat -Wformat-security")
 # set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wconversion -Wsign-conversion")
 
 if (WAMR_BUILD_TARGET MATCHES "X86_.*" OR WAMR_BUILD_TARGET STREQUAL "AMD_64")
-  if (NOT (${CMAKE_C_COMPILER} MATCHES ".*clang.*"))
+  if (NOT (CMAKE_C_COMPILER_ID STREQUAL "Clang"))
```

Comment:
CMAKE_C_COMPILER_ID may be:
  AppleClang,  means Apple Clang (apple.com)
  Clang, means LLVM Clang (clang.llvm.org)
Seems it is better to use MACHES "*Clang" but not  STREQUAL.
Also the CMAKE_C_COMPILER_ID may be unuset if we set (CMAKE_C_COMPILER "clang") in this file.
So we had better check both of them:
  if (NOT (CMAKE_C_COMPILER MATCHES ".*clang.*" OR CMAKE_C_COMPILER_ID MATCHES ".*Clang"))